src/llmq/instantsend.cpp

```diff
@@ -1526,7 +1526,7 @@ void CInstantSendManager::AskNodesForLockedTx(const uint256& txid, const CConnma
         if (nodesToAskFor.size() >= 4) {
             return;
         }
-        if (!pnode->m_block_relay_only_peer) {
```

Comment:
Upstream PR doesn't change any invocation of `m_block_relay_only_peer` why'd you do so here in 17164?

Reply:
Because:
```
// in bitcoin: m_tx_relay == nullptr if we're not relaying transactions with this peer
// in dash: m_tx_relay should never be nullptr, use m_block_relay_only_peer == true instead
    std::unique_ptr<TxRelay> m_tx_relay;
```
Due to #4888

Reply:
can we pull this backport into it's own PR? It feel more like a commit inspired by bitcoin's PR rather than just backporting it. And feels more likely to potentially cause issues